packages/rooks/src/hooks/useFullscreen.ts

```diff
@@ -1,266 +1,226 @@
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-nocheck
-import { UnknownFunction } from "@/types/utils";
-import { useState, useCallback, useRef } from "react";
-import { useDocumentEventListener } from "./useDocumentEventListener";
-import { warning } from "./warning";
-
-type EventCallback = (this: Document, event_: unknown) => unknown;
-type OnChangeEventCallback = (
-  this: Document,
-  event_: unknown,
-  isOpen: boolean
-) => unknown;
-
-type NormalizedFullscreenApi = {
-  exitFullscreen: string;
-  fullscreenElement: string;
-  fullscreenEnabled: string;
-  fullscreenchange: string;
-  fullscreenerror: string;
-  requestFullscreen: string;
-};
-
-const getFullscreenControls = (): NormalizedFullscreenApi => {
-  const functionMap = [
-    [
-      "requestFullscreen",
-      "exitFullscreen",
-      "fullscreenElement",
-      "fullscreenEnabled",
-      "fullscreenchange",
-      "fullscreenerror",
-    ],
-    // New WebKit
-    [
-      "webkitRequestFullscreen",
-      "webkitExitFullscreen",
-      "webkitFullscreenElement",
-      "webkitFullscreenEnabled",
-      "webkitfullscreenchange",
-      "webkitfullscreenerror",
-    ],
-    // Old WebKit
-    [
-      "webkitRequestFullScreen",
-      "webkitCancelFullScreen",
-      "webkitCurrentFullScreenElement",
-      "webkitCancelFullScreen",
-      "webkitfullscreenchange",
-      "webkitfullscreenerror",
-    ],
-    [
-      "mozRequestFullScreen",
-      "mozCancelFullScreen",
-      "mozFullScreenElement",
-      "mozFullScreenEnabled",
-      "mozfullscreenchange",
-      "mozfullscreenerror",
-    ],
-    [
-      "msRequestFullscreen",
-      "msExitFullscreen",
-      "msFullscreenElement",
-      "msFullscreenEnabled",
-      "MSFullscreenChange",
-      "MSFullscreenError",
-    ],
-  ];
-
-  const returnValue = {} as NormalizedFullscreenApi;
-
-  for (const functionSet of functionMap) {
-    if (functionSet && functionSet[1] in document) {
-      for (const [index, _function] of functionSet.entries()) {
-        returnValue[functionMap[0][index]] = functionSet[index];
-      }
-    }
+import { useState, useCallback, useEffect } from "react";
+import type { RefObject } from "react";
+
+declare global {
+  interface Element {
+    webkitRequestFullscreen: Element["requestFullscreen"];
```

Comment:
```ts
    webkitRequestFullscreen?: Element["requestFullscreen"];
```

Not sure if necessary, but how about making all of them undefined-able just to make it clear it is possibly undefined depending on the browser?

Reply:
I was thinking about it. I couldnt find a way to replace existing type definitions (I wanted to make non-prefixed methods also optional). I didn't want to mix them. Seems more uniform this way. If there is a way to override it I can make further adjustments. Let me know what you think.

Reply:
I was also thinking about creating a separate type and casting document to it. Might be a better solution but I'm not sure if you like it.

Reply:
I'm also not sure how we would extend the event map that way.

Reply:
I would generally prefer creating a separate type, but I have no idea what will that look like...

Reply:
I'll try to provide some examples in following days. Can @imbhargav5 start working on tests? Types should not interfere.

Reply:
Resolved in latest commit. I think it didn't turn out that bad.